include/real/real_exception.hpp

```diff
@@ -112,7 +112,11 @@ namespace boost {
             }
         };
        
-
+        struct logarithm_not_defined_for_non_positive_number : public std::exception {
+            const char * what() const throw () override {
+                return "Logarithm is only defined for positive number";
+            }
```

Comment:
Other logarithm functions in the STL do not provide a exception. 
This is some difference in behavior that will block it from been a drop in replacement for float for example. 
The expected result here would be "NaN".
I think we should have an overload of numeric_limits<Real> that includes how to define NaN.

Reply:
I agree with that, I would add the issue in the github, but I would care for know.